lib/server.js

```diff
@@ -37,31 +37,57 @@ module.exports = class AlgoServer extends EventEmitter {
     } = args
 
     this.port = port
-    this.host = new AOHost({
-      aos, apiKey, apiSecret, agent, wsURL, restURL
-    })
+    // If provided with order definitions then create the algoHost instance
+    // Mostly just for backwards compatibility
+    if (aos) {
+      host = new AOHost({
+        aos, apiKey, apiSecret, agent, wsURL, restURL
+      })
+      this.setAlgoHost(host)
+    }
+
+    this.hbi = startHB(this)
+    this.wss = startWSS(this)
+  }
+
+  close () {
+    this.host.close()
+    this.wss.close()
+  }
 
+  /**
+   * Sets the event bindings to use the given algo host instance. This function
+   * also manages the closing of the old aoHos instance and the establishment of the
+   * new algohost instance.
+   * @param {Object} aoHost
+   */
+  setAlgoHost (aoHost) {
+    if (this.host) {
+      // remove all events from old host
+      this.host.removeAllListeners('ao:start')
```

Comment:
Can you move this into a `clearAlgoHost()` method or something similar? Better yet, do so & extract the listener strings into a constant at the top of the file, so we can just loop over them for adding/removing listeners.

Reply:
Good idea, found a better way to do it:

```
this.host.eventNames().forEach(eventName => this.host.removeAllListeners(eventName))
```